crates/optimism/rpc/src/eth/pending_block.rs

```diff
@@ -3,16 +3,40 @@
 use std::sync::Arc;
 
 use crate::{OpEthApi, OpEthApiError};
-use alloy_eips::BlockNumberOrTag;
-use reth_primitives_traits::RecoveredBlock;
+use alloy_consensus::{BlockHeader, Transaction};
+use alloy_eips::eip7840::BlobParams;
+use alloy_json_rpc::{Id, Request, SerializedRequest};
+use alloy_pubsub::PubSubConnect;
+use alloy_rpc_client::WsConnect;
+use alloy_rpc_types_mev::FlashBlock;
+use jsonrpsee_core::Cow;
+use reqwest::Url;
+use reth_chainspec::{ChainSpecProvider, EthChainSpec};
+use reth_errors::{BlockExecutionError, BlockValidationError, RethError};
+use reth_evm::{
+    execute::{BlockBuilder, BlockBuilderOutcome, ExecutionOutcome},
+    ConfigureEvm, Evm,
+};
+use reth_primitives_traits::{
+    transaction::error::InvalidTransactionError, Recovered, RecoveredBlock, SignedTransaction,
+};
+use reth_revm::{database::StateProviderDatabase, State};
 use reth_rpc_eth_api::{
     helpers::{pending_block::PendingEnvBuilder, LoadPendingBlock},
-    FromEvmError, RpcConvert, RpcNodeCore,
+    FromEthApiError, FromEvmError, RpcConvert, RpcNodeCore,
 };
-use reth_rpc_eth_types::{EthApiError, PendingBlock};
-use reth_storage_api::{
-    BlockReader, BlockReaderIdExt, ProviderBlock, ProviderReceipt, ReceiptProvider,
+use reth_rpc_eth_types::{
+    utils::recover_raw_transaction, EthApiError, PendingBlock, PendingBlockEnvOrigin,
 };
+use reth_storage_api::{ProviderBlock, ProviderReceipt, ReceiptProvider, StateProviderFactory};
+use reth_transaction_pool::{
+    error::InvalidPoolTransactionError, BestTransactionsAttributes, TransactionPool,
+};
+use revm::context::Block;
+use serde_json::Value;
+use tokio::time::Duration;
+
+pub(super) type SharedFlashblockState = Arc<tokio::sync::RwLock<FlashBlock>>;
```

Comment:
prefer if we don't have this alias, so that it is easier to spot where locks are being used inline to make it easier to debug (locks are statistically bug prone sync primitive compared to others)